cursor.c

```diff
@@ -335,7 +344,11 @@ int php_mongocursor_advance(mongo_cursor *cursor TSRMLS_DC)
 	/* Free the previous current item */
 	php_mongo_cursor_clear_current_element(cursor);
 
-	cursor->at++;
+	if (cursor->cursor_options & MONGO_CURSOR_OPT_DONT_ADVANCE_ON_FIRST_NEXT) {
+		cursor->cursor_options ^= MONGO_CURSOR_OPT_DONT_ADVANCE_ON_FIRST_NEXT;
```

Comment:
I was going to ask about when this flag was removed.

Is it still possible that it might persist if we created a cursor, called `hasNext()` (thereby setting this flag), and then never accessed the element or called `next()`/`getNext()`? On that cursor, if we then reset it and started iterating as normal with a foreach, would we miss the first advance?